pkg/sync/task/queue/queue.go

```diff
@@ -56,19 +56,20 @@ func (q *Queue) popQueue() func() {
 
 // Submit appends and execute task by Queue.
 func (q *Queue) Submit(tasks ...func()) error {
-	err := q.appendQueue(tasks)
-	if err != nil {
+	if err := q.appendQueue(tasks); err != nil {
 		return err
 	}
-	q.goPool.Submit(func() {
+	if err := q.goPool.Submit(func() {
 		for {
 			task := q.popQueue()
 			if task == nil {
 				return
 			}
 			task()
 		}
-	})
+	}); err != nil {
+		return err
+	}
```

Comment:
we just try to submit action - so when it fails, it means that we reach the limit of goroutines, and the task will be processed by another goroutine.

pls change it to: `_ = q.goPool.Submit(...`